### PR TITLE
Feature/230

### PR DESCRIPTION
**Pull request by adrijanik**
#### Related Issue(s)

Issue #230 
#### Description of Changes

Added parameter `tensorboard=False` to model fit function (EmbeddingModel derivatives) to visualize average training loss after each epoch with tensorboard.
After model is trained with e.g. CompleX(...,tensorboard=True), tensorboard_files folder is created where loss is being logged. To see the loss, run: 
`tensorboard --logdir tensorboard_files/`
and go to your browser to see the loss plotted.

### CODE REVIEW

File: ampligraph/latent_features/models/EmbeddingModel.py [hunk 1]
```diff
@@ -944,7 +944,7 @@ def _training_data_generator(self):
             else:
                 yield np.squeeze(out_triples), unique_entities, entity_embeddings
 
-    def fit(self, X, early_stopping=False, early_stopping_params={}, focusE_numeric_edge_values=None):
+    def fit(self, X, early_stopping=False, early_stopping_params={}, focusE_numeric_edge_values=None, tensorboard=False):
```
Comment (sumitpai) [inline ampligraph/latent_features/models/EmbeddingModel.py:947]: I would suggest replacing this with tensorboard_logs_path =None. If none then dont save tf summary, if not save it at desired location

File: ampligraph/latent_features/models/EmbeddingModel.py [hunk 2]
```diff
@@ -1169,13 +1177,19 @@ def fit(self, X, early_stopping=False, early_stopping_params={}, focusE_numeric_
                         pass
 
                     if self._perform_early_stopping_test(epoch):
+                        if tensorboard:
+                            self.writer.flush()
```
Comment (sumitpai) [inline ampligraph/latent_features/models/EmbeddingModel.py:1181]: I would suggest plotting validation loss, and validation mrr as well (in perform_early_stopping_test)

File: ampligraph/latent_features/models/EmbeddingModel.py (hunk 1 above)
Comment (adrijanik) [reply ampligraph/latent_features/models/EmbeddingModel.py:947]: Good point, will add that thanks.

File: ampligraph/latent_features/models/EmbeddingModel.py (hunk 2 above)
Comment (adrijanik) [reply ampligraph/latent_features/models/EmbeddingModel.py:1185]: Validation loss can't be added in this release as it is batch number dependent, will be added in AmpliGraph2. Criteria for early stopping added to the tracking.
